src/jit/lsraarmarch.cpp

```diff
@@ -738,8 +738,22 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTreePutArgStk* argNode, fgArgTabEntr
     else
     {
 #ifdef _TARGET_ARM_
+
+#ifndef ARM_SOFTFP
         // We must not have a multi-reg struct; double uses 2 slots and isn't a multi-reg struct
         assert((info->numSlots == 1) || ((info->numSlots == 2) && (putArgChild->TypeGet() == TYP_DOUBLE)));
+#else  // ARM_SOFTFP
+        // The `double` types have been transformed to `long` on armel.
+        if ((info->numSlots == 2) && (putArgChild->TypeGet() == TYP_LONG))
+        {
+            argNode->gtLsraInfo.srcCount = 2;
+        }
+        else
+        {
+            assert(info->numSlots == 1);
+        }
+#endif // ARM_SOFTFP
+
```

Comment:
As a rule, when you have both an if and else clause, you should refactor to avoid the double negative (i.e. in this case put the `ARM_SOFTFP` code first). However, the above could be greatly simplified as:
```
#ifdef ARM_SOFTFP
        // The `double` types have been transformed to `long` on armel.
        var_types doubleType = TYP_LONG;
#else
        var_types doubleType = TYP_DOUBLE:
#endif
        assert((info->numSlots == 1) || ((info->numSlots == 2) && (putArgChild->TypeGet() == doubleType)));
```